nixos/modules/services/misc/gitlab.nix

```
@@ -1238,7 +1238,7 @@ in
         message = "services.gitlab.secrets.activeRecordSaltFile must be set!";
       }
       {
-        assertion = versionAtLeast postgresqlPackage.version "16";
+        assertion = cfg.databaseCreateLocally -> (versionAtLeast postgresqlPackage.version "16");
```

```suggestion
        assertion = (cfg.databaseHost == "") -> (versionAtLeast postgresqlPackage.version "16");
```

I think a better approach would be to check whether `cfg.databaseHost` had been modified. There could be a scenario where a user opts to provisioning a database themself by setting `databaseCreateLocally` to `false`.